File: src/test/scala/anti/patterns/PriceEngineTests.scala
package anti.patterns

import demo.customer.{PriceEngine, Product}
import org.scalatest.flatspec.AnyFlatSpec

class PriceEngineTests extends AnyFlatSpec {
  it should "calculate the discount of 2 products" in {
    val products = List(Product("P1"), Product("P2"), Product("P3"))

    val discount = PriceEngine.calculateDiscount(products: _*)

    assert(products.length * 0.01 == discount)
  }
}

File: csharp/CleanTests.Tests/AntiPatterns/Solution/PriceEngineTests.cs
using CleanTests.Customer;
using FluentAssertions;
using LanguageExt;
using Xunit;

namespace CleanTests.Tests.AntiPatterns.Solution;

public class PriceEngineTests
{
    [Fact]
    public void Discount_Of_3_Products_Should_Be_3_Percent()
    {
        var products = Seq.create(new Product("P1"), new Product("P2"), new Product("P3"));
        var discount = PriceEngine.CalculateDiscount(products.ToArray());

        discount.Should().Be(0.03);
    }
}